components/switch/xiaomi.py

```diff
@@ -54,15 +56,11 @@ def is_on(self):
 
     def turn_on(self, **kwargs):
         """Turn the switch on."""
-        if self.xiaomi_hub.write_to_hub(self._sid, self._data_key, 'on'):

```

Comment:
Because after turn_on and turn_off is called, parse_data will be called immediately. parse_data function will update switch _state and update _state value to HASS if it change